AdyenDropIn/Modules/Component/ComponentContainerAssembler.swift

```diff
@@ -9,20 +9,51 @@ import Foundation
 import UIKit
 
 internal protocol ComponentContainerAssemblerProtocol {
-    func resolveContainerView(for component: PresentableComponent) -> UIViewController
+    func resolveComponentContainerRouter(
+        for component: PresentableComponent,
+        delegate: ComponentContainerRouterDelegate
+    ) -> ComponentContainerRouterProtocol
 }
 
 internal struct ComponentContainerAssembler: ComponentContainerAssemblerProtocol {
 
+    // MARK: - Properties
+
+    private let context: AdyenContext
+    private let configuration: DropInComponent.Configuration
+    private let cardComponentDelegate: CardComponentDelegate?
+    private let partialPaymentDelegate: PartialPaymentDelegate?
+
+    // MARK: - Initializers
+
+    internal init(
+        context: AdyenContext,
+        configuration: DropInComponent.Configuration,
+        cardComponentDelegate: CardComponentDelegate?,
```

Comment:
I didn't find any usages for this being optional.
I'd rather ditch struct here and go with classes composition or simple inheritance to use different instances depending on the need:
- `ComponentContainerAssembler`
- `CardComponentContainerAssembler`

Maybe I'm missing a bigger picture, but from this PR it looks like `cardComponentDelegate` is something very specific that leaks into an abstract interface. Probably just removing this scpecific delegate from the interface and making it optional would allow to hide the assigning logic in the scope when it matters and makes sense.

Same applies to `PartialPaymentDelegate` type